### ISSUE TITLE
ManifoldSAE backprop-only training learns degenerate atom routing (encoder amplitude ~uncorrelated with features; stable across 800-8000 steps)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#583]
## Summary

When `gamfit.torch.ManifoldSAE` is trained by **backprop only** (no closed-form
`sae.fit()` co-training — which is currently the only option, since `fit()` is
broken: #577 / #578), the `softmax_topk` encoder **fails to learn atom
assignment**: on a clean synthetic task where the ground-truth structure is
known, most planted features end up with ~zero correlation between any atom's
amplitude and that feature's presence. The decoder still reconstructs (EV ~0.77)
via distributed/entangled atoms, but the one-atom-per-manifold routing — the
whole point of a Manifold-SAE — does not emerge. More training does not fix it.

## Setup

4 distinct manifolds (line, parabola, half-arc, circle) planted in **mutually
orthogonal** subspaces of R^32 (unambiguous ground truth), sparse sum, top_k=2,
6 atoms (4 + 2 slack). After training, measure Pearson correlation between each
atom's amplitude and each GT feature's active indicator; for each feature take
the best atom.

## Evidence — degeneracy is stable across training length

```
steps=  800  best_atom_corr_per_feat=[-0.04, -0.07, 0.49, 0.15]  distinct_atoms=4/4  pos_std=0.220
steps= 3000  best_atom_corr_per_feat=[ 0.02, -0.02, 0.48, 0.08]  distinct_atoms=3/4  pos_std=0.295
steps= 8000  best_atom_corr_per_feat=[ 0.03, -0.01, 0.48, 0.15]  distinct_atoms=4/4  pos_std=0.320
```

Features 0 (line) and 1 (parabola) are **never** detected by any atom (best
correlation ~0 / negative) at any training length; only the half-arc routes
reliably (~0.48). The full atom×feature correlation matrix shows several atoms
piling onto the same feature while others stay silent.

The curve math itself is fine: fitting `gaussian_reml_fit_positions` to *clean*
single-feature targets recovers each planted orthogonal subspace at
principal-angle cos ≈ 1.00 and leakage ≈ 0.00. So the failure is specifically the
**encoder's assignment/amplitude**, not the decoder spline fit.

## Repro

```python
import numpy as np, torch
from manifold_sae.data_synthetic import SyntheticDataset   # planted orthogonal manifolds
from gamfit.torch import (ManifoldSAE, ManifoldSAEConfig, DecoderConfig,
                          SparsityConfig, RemlConfig)
from torch.utils.data import DataLoader

ci=[0,1,8,9]; F=4; D=32; NS=F+2
torch.manual_seed(0); np.random.seed(0)
ds=SyntheticDataset(d_ambient=D,n_features=F,n_samples=3000,sparsity=0.25,
                    noise=0.02,seed=0,orthogonal_subspaces=True,curve_indices=ci)
cfg=ManifoldSAEConfig(input_dim=D,n_atoms=NS,intrinsic_rank=2,atom_manifold="product",
        n_basis_per_atom=10,sparsity=SparsityConfig(kind="softmax_topk",target_k=2),
        decoder=DecoderConfig(ortho_weight=1e-2),reml=RemlConfig())
sae=ManifoldSAE(cfg); opt=torch.optim.Adam(sae.parameters(),lr=2e-3)
loader=DataLoader(ds,batch_size=128,shuffle=True,drop_last=True)
# ... standard backprop loop: out=sae(x); loss=mse+coverage+ortho; loss.backward() ...
# (SyntheticDataset lives in github.com/SauersML/Manifold-SAE; any sparse-sum-of-
#  orthogonal-curves generator reproduces it.)

active=ds.ground_truth["active"].numpy().astype(float)
amp=sae(ds.x.to(cfg.dtype)).amplitudes.detach().numpy()
C=np.corrcoef(np.c_[amp,active].T)[:NS,NS:]   # atom x feature
print([round(C[:,i].max(),2) for i in range(F)])   # best atom per feature ~ [0, 0, 0.48, 0.1]
```

## Questions / asks

1. Is there an **intended training recipe** for `ManifoldSAE` when the closed-form
   `fit()` co-training is unavailable? Right now backprop-only does not discipline
   the encoder into one-atom-per-feature.
2. Should the `softmax_topk` assignment have a stronger feature-selection
   gradient (the amplitudes look nearly decoupled from which features are
   present)? An assignment-entropy / load-balancing term, or temperature
   annealing, may be missing from the public path.
3. This is likely entangled with #577 / #578 — if the closed-form solve worked,
   it would co-train the decoder and (presumably) pull the encoder into clean
   routing. Confirming that would be useful.

## Environment
gamfit 0.1.141, torch 2.12.0, CPU, macOS arm64.

**Comment by SauersML** [comment]
## Investigation conclusion — not a code defect in the assignment path

Read the torch encoder/assignment (`gamfit/torch/manifold_sae.py`): the
`softmax_topk` path is a standard straight-through estimator —
`probs = softmax(amp_logits/τ)`, top-k mask, `assignments = probs + (hard −
probs).detach()`, `amp = softplus(amp_logits)`. No defect there.

The degenerate routing (encoder amplitude ~uncorrelated with planted features,
stable 800→8000 steps) is a **training-dynamics** outcome of backprop-only
training, which is the only available mode while the closed-form `sae.fit()`
co-training is unusable (#577 fatal rejection — now fixed — and the deeper Efs
runtime cost; plus #578). The closed-form solve is the mechanism intended to
discipline atom assignment; with it down, pure backprop at small scale does not
learn one-atom-per-manifold.

So #583 is effectively downstream of #577/#578: the right next step is to make
`sae.fit()` cheap enough to co-train (the fast-path delegation suggested in
#577), then re-measure routing with `fit_every>0`. A speculative assignment
change (lower `τ_min`, load-balancing/entropy term) might help but needs
validation that's gated on the same slow fit. Leaving this open as
training-dynamics rather than a one-line bug.

**Comment by SauersML** [comment]
🔒 claim: claude-code coordinator session — taking this now. (cluster: sae-training)
